Trappist/src/Promact.Trappist.Web/wwwroot/app/tests/test-questions/test-questions.component.ts

```diff
@@ -162,26 +173,43 @@ export class TestQuestionsComponent implements OnInit {
      * @param questionCount number of all the questions of a category
      */
     selectAll(category: Category) {
-        category.questionList.map(function (questionList) {
-            //If selectAll checkbox is selected
-            if (category.selectAll) {
-                //every question is selected
-                questionList.question.isSelect = true;
-                category.numberOfSelectedQuestion = category.questionCount;
-            }
-            else {
-                //If selectAll checkbox is unselected ,then every question is deselected
-                questionList.question.isSelect = false;
-                category.numberOfSelectedQuestion = 0;
-            }
-        });
+            category.questionList.map(function (questionList) {
+                //If selectAll checkbox is selected
+                if (category.selectAll) {
+                    //every question is selected
+                    questionList.question.isSelect = true;
+                    category.numberOfSelectedQuestion = category.questionCount;
+                }
+                else {
+                    //If selectAll checkbox is unselected ,then every question is deselected
+                    questionList.question.isSelect = false;
+                    category.numberOfSelectedQuestion = 0;
+                }
+            });
     }
 
     /**
      * Adds the questions to question table and redirect to test dashboard
      */
     saveExit() {
         this.isSaveExit = true;
-        this.saveNext();
+        if (this.isEditTestEnabled)
+            this.saveNext();
+        else
+            this.route.navigate(['/tests']);
+    }
+
+    /**
+    * Checks if any candidate has taken the test
+    */
+    isTestAttendeeExist() {
+        this.testService.isTestAttendeeExist(this.testId).subscribe((response) => {
+            if (response.response) {
```

Comment:
response.response does not make any sense. if it returns boolean value then this code can be optimized.